src/language/templates/visitors/checkparent_visitor.hpp

```diff
@@ -29,20 +27,36 @@ namespace visitor {
  */
 
 /**
- * \brief Visitor that checks the parents
+ * \class JSONVisitor
+ * \brief %Visitor for checking parents of ast nodes
```

Comment:
As @pramodk suggested, I would expand a little on explaining the way this visitor works (for future a future colleague), also fix doxygen class name please.